### PR TITLE
Make this a python package.

### PR DESCRIPTION
**Pull request by isuruf**
`conda-forge-tick --run 0` (or 1 or 2) works for now

### CODE REVIEW

File: conda_forge_tick/check_new_version.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        print(node, attrs['version'], attrs['new_version'])
+
+    print('Current number of out of date packages not PRed: {}'.format(
+        str(len([n for n, a in gx.node.items()
+                 if a['new_version']  # if we can get a new version
+                 and a['new_version'] != a['version']  # if we need a bump
+                 and a.get('PRed', '000') != a['new_version']  # if not PRed
+                 ]))))
+    print('writing out file')
+# nx.write_yaml(gx, 'graph.yml')
```
Comment (CJ-Wright) [inline conda_forge_tick/check_new_version.py]: Do these need to be here?

File: conda_forge_tick/update_upstream_versions.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import requests
+from pkg_resources import parse_version
+import feedparser
+from .utils import parsed_meta_yaml
+
+import logging
+
+logger = logging.getLogger("conda_forge_tick.update_upstream_versions")
+
+class VersionFromFeed:
```
Comment (CJ-Wright) [inline conda_forge_tick/update_upstream_versions.py]: Should these been pulled from the library?

File: conda_forge_tick/update_upstream_versions.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                    meta_yaml['name'], source.name))
+    with open('upstream_bad', 'a') as f:
+        f.write('{}: unknown source\n'.format(meta_yaml['name']))
+    return False
+
+
+def update_upstream_versions(gx, sources=[PyPI(), CRAN(), Github()]):
+
+    for node, attrs in gx.node.items():
+        attrs['new_version'] = get_latest_version(attrs, sources)
```
Comment (CJ-Wright) [inline conda_forge_tick/update_upstream_versions.py]: If the version is coming from CRAN would it be possible to flip the version back to something that the CRAN upstream understands? (There were some issues finding the right tarballs last round)

File: conda_forge_tick/all_feedstocks.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+def get_all_feedstocks(cached=False):
+    if cached:
+        logger.info('reading names')
+        with open('names.txt', 'r') as f:
+            names = f.read().split()
+        return names
+
+    names = get_all_feedstocks_from_github()
+    with open('names.txt', 'w') as f:
```
[the remark this answers is not in the document]
Comment (isuruf) [reply conda_forge_tick/all_feedstocks.py]: I think we don't need two separate scripts 0 and 1. We can merge them together.

[the remark this answers is not in the document]
Comment (CJ-Wright) [reply conda_forge_tick/all_feedstocks.py]: That is most likely true, now we don't run into the API limit on 1.

File: conda_forge_tick/update_upstream_versions.py (hunk 2 above)
Comment (isuruf) [reply conda_forge_tick/update_upstream_versions.py]: I don't understand what you are trying to say here

Comment (CJ-Wright) [reply conda_forge_tick/update_upstream_versions.py]: Sorry I had things backwards, it seems there are duplicates between this file and https://github.com/regro/cf-scripts/pull/85/files#diff-94f6e88874a59b9d037066a48e6b6c3fR11

File: conda_forge_tick/update_upstream_versions.py [hunk 5]
```diff
@@ -88,28 +131,39 @@ def get_latest_version(meta_yaml):
...
     with open('upstream_bad', 'a') as f:
-        f.write('{}: unknown source at {}\n'.format(meta_yaml['name'],
-                                                    meta_yaml['url']))
+        f.write('{}: unknown source\n'.format(meta_yaml['name']))
     return False
 
 
-# gx = nx.read_yaml('graph.yml')
-gx = nx.read_gpickle('graph.pkl')
+def update_upstream_versions(gx, sources=[PyPI(), CRAN(), Github()]):
```
Comment (CJ-Wright) [inline conda_forge_tick/update_upstream_versions.py]: Should `RawURL` be in the sources? Also can the sources be a tuple (so we can avoid unintended mutation)?

File: conda_forge_tick/update_upstream_versions.py [hunk 6]
```diff
@@ -2,6 +2,11 @@
 import requests
 from pkg_resources import parse_version
 import feedparser
+from .utils import parsed_meta_yaml
```
Comment (CJ-Wright) [inline conda_forge_tick/update_upstream_versions.py]: do we also need to import `NullUndefined`?

**Comment by CJ-Wright** [comment]
I think this is good to go, although I think there is an issue with the CRAN version number replacement (from what I can tell from the CRAN page, there are a few packages which have `-` in their version number eg BIFIEsurvey_2.3-18.tar.gz at https://cran.r-project.org/src/contrib/)
I'd like to hold off on merging this until Monday so I can reset the Travis setup.

@scopatz and @justcalamari thoughts?

File: conda_forge_tick/update_upstream_versions.py [hunk 7]
```diff
@@ -75,10 +80,46 @@ def get_version(self, url):
         return str(ver).replace('-', '_')
 
 
-sources = [PyPI(), CRAN(), Github()]
+
+class RawURL:
+    name = 'RawURL'
+    def get_url(self, meta_yaml):
+        pkg = meta_yaml['name']
```
Comment (justcalamari) [inline conda_forge_tick/update_upstream_versions.py]: As @CJ-Wright mentioned in #83, this should be `pkg = meta_yaml['feedstock_name']`.

File: conda_forge_tick/make_graph.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                'build', []) is not None else [])
+            run = list(req.get('run', []) if req.get(
+                'run', []) is not None else [])
+            req = build + run
+            req = set([x.split()[0] for x in req])
+
+        if not ('url' in yaml_dict.get('source', {})
+                and 'name' in yaml_dict.get('package', {})
+                and 'version' in yaml_dict.get('package', {})):
+            logger.warn("Recipe {} doesn't have a url".format(name))
```
Comment (justcalamari) [inline conda_forge_tick/make_graph.py]: Here the problem was that the recipe was missing url, name, or version, but we do not know which after the if statement.

**Pull request merged**